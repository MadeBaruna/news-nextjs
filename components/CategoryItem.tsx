import { StatelessComponent } from 'react';
import styled from 'styled-components';
import { withRouter, WithRouterProps } from 'next/router';
import Link from 'next/link';

const Item = styled.span<{ active: boolean }>`
  font-family: "Raleway", sans-serif;
  font-size: 18px;
  padding: 0 8px;
  user-select: none;
  color: ${({ active }) => (active ? 'red' : 'black')};

  &:hover {
    color: red;
  }

  &:active {
    color: red;
  }
`;

interface IProps extends WithRouterProps {
  name: string;
  category: string;
}

const CategoryItem: StatelessComponent<IProps> = ({
  name,
  router,
  category,
}) => (
  <Link href={`/?category=${category}`} as={`/category/${category}`}>
    <Item
      active={
        (category === 'general' && router.query.category === undefined) ||
        router.query.category === category
      }
    >
      {name}
    </Item>
  </Link>
);

export default withRouter(CategoryItem);
