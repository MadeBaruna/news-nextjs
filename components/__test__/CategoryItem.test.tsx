import { shallow } from 'enzyme';
import React from 'react';
import CategoryItem from '../CategoryItem';

describe('Header', () => {
  it('Render Header', () => {
    const wrapper = shallow(<CategoryItem name="General" category="general" />);
    expect(wrapper).toMatchSnapshot();
  });
});
