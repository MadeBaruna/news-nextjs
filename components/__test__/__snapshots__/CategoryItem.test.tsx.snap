// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem Render CategoryItem 1`] = `ShallowWrapper {}`;
